### ISSUE TITLE
GeoJSON coordinates change when tiles are overscaled

### ISSUE THREAD

**Issue by sillfishes** [mapbox/mapbox-gl-js#1205]
When a layer is zoomed greater than z16, the coordinates of the features of the layer seem to be either reprojected or scaled.  Here's the same point on the map zoomed to three different levels:

![screen shot 2015-05-15 at 9 27 19 am](https://cloud.githubusercontent.com/assets/6751140/7653615/17d2b346-fae5-11e4-81a5-92fc5c8da90b.png)

![screen shot 2015-05-15 at 9 27 49 am](https://cloud.githubusercontent.com/assets/6751140/7653617/1a658b24-fae5-11e4-887c-f1e1d7eceb86.png)

![screen shot 2015-05-15 at 9 28 05 am](https://cloud.githubusercontent.com/assets/6751140/7653620/1ce2d992-fae5-11e4-900f-bfdde52f7771.png)

Is this expected behavior?  Without being able to override the zoom level of the featuresAt function, seems like there's no way to get the original coordinates of a feature that match the CRS of the map.  If this is expected behavior, is there a way to roll back the scaling so that the original coordinates of the feature can be retrieved?

**Comment by jfirebaugh** [comment]
No, this is a bug. What version of mapbox-gl-js are you using? It may have been something that was fixed already in the master branch.

**Comment by sillfishes** [comment]
Thanks for confirming.  We're using 0.7.0.

**Comment by PabloGancharov** [comment]
what I found is that the error is at: https://github.com/mapbox/vector-tile-js/blob/master/lib/vectortilefeature.js line 125

```

VectorTileFeature.prototype.toGeoJSON = function(x, y, z) {
    var size = this.extent * Math.pow(2, z),
        x0 = this.extent * x,
        y0 = this.extent * y,
        coords = this.loadGeometry(),
        type = VectorTileFeature.types[this.type];

    for (var i = 0; i < coords.length; i++) {
        var line = coords[i];
        for (var j = 0; j < line.length; j++) {
            var p = line[j], y2 = 180 - (p.y + y0) * 360 / size;
            line[j] = [
                (p.x + x0) * 360 / size - 180,
                360 / Math.PI * Math.atan(Math.exp(y2 * Math.PI / 180)) - 90
            ];
        }
    }

    if (type === 'Point' && coords.length === 1) {
        coords = coords[0][0];
    } else if (type === 'Point') {
        coords = coords[0];
        type = 'MultiPoint';
    } else if (type === 'LineString' && coords.length === 1) {
        coords = coords[0];
    } else if (type === 'LineString') {
        type = 'MultiLineString';
    }

    return {
        type: "Feature",
        geometry: {
            type: type,
            coordinates: coords
        },
        properties: this.properties
    };
};

```

when parameter z gets  than 16,  the result is wrong

**Comment by jfirebaugh** [comment]
@ansis This looks like an overscaling bug. Was it fixed already in 2cfd9d0687e430d92c62f6ee2cbd5fd91d8fb44e?

**Comment by jorgesierra** [comment]
Hi guys, any news about issue? Thanks!

**Comment by ansis** [comment]
thanks for tracking down the problem. fixed in e3a644c

**Comment by MichalNemec** [comment]
Still having this issue at version 1.15.2
